ecg_segmentation_dataset.py

```diff
@@ -0,0 +1,35 @@
+import torch
+import einops
+from torch.utils.data import Dataset
+from datasets import Value, Array2D, Features, load_dataset
+
+
+class ECGDataset(Dataset):
+    def __init__(self, huggingface_path: str):
+        super().__init__()
+
+        features = Features(
+            {
+                "record_id": Value(dtype="string"),
+                "signal": Array2D(dtype="float32", shape=(1000, 2)),
+                "mask": Array2D(dtype="int8", shape=(1000, 1)),
+            }
+        )
+        self.data = load_dataset(huggingface_path, features=features, split="train")
+
+    def __len__(self):
+        return len(self.data)
+
+    def __getitem__(self, index: int) -> tuple[torch.Tensor, torch.Tensor]:
+        record = self.data[index]
+
+        # wrap signal and mask to torch.Tensor
+        signal = torch.tensor(record["signal"], dtype=torch.float32)
+        mask = torch.tensor(record["mask"], dtype=torch.float32)
+
+        # reshape data: [sequence_length, num_channels] -> [num_channels, sequence_length]
+        signal = einops.rearrange(signal, "l c -> c l")
+        # loss
+        mask = einops.rearrange(mask, "l c -> c l")
+
+        return signal, mask
```

Comment:
```suggestion
        record = self.data[index]

        # wrap signal and mask to torch.Tensor
        signal = torch.tensor(record["signal"], dtype=torch.float32)
        mask = torch.tensor(record["mask"], dtype=torch.float32)

        # reshape data: [sequence_length, num_channels] -> [num_channels, sequence_length]
        signal = einops.rearrange(signal, "l c -> c l")
        # loss
        mask = einops.rearrange(mask, "l c -> c l")

        item = {
          "signal": signal,
          "mask": mask,
          "record_id": record["record_id"],
        }
        return item
```

We may need the record id for better validation strategies. You'd also need to change the way batches are unpacked to something like this:

```python
for batch in loader:
  signal = batch["signal"]
  mask = batch["mask"]
```